tests/validate_test.rs

```diff
@@ -35,12 +35,10 @@ packs/foo, packs/bar",
 #[test]
 fn test_validate_architecture() -> Result<(), Box<dyn Error>> {
     let expected_message_1 = String::from(
-        "
```

Comment:
I think we still want the architecture validator to ensure that stated dependencies do not violate the architecture right?

Is your point that if there is a reference that the checker finds, then it's the reference that violates architecture – not the stated dependencies?  And we already have that behavior in the system?

My only thought here is that even if there are zero references, it's still wrong to have a dependency that conceptually violates a checker that should be restricting that dependency from existing. Can you share more about why you feel like this should be removed?

Reply:
Thanks for the feedback! There's a few concerns we have about one type of validation using another type's package.yml configuration.

1) Maintaining separation between dependency definitions and architectural layer validations is beneficial. This separation allows teams to selectively apply different types of validations without worrying about interdependencies between them.
2) If layer validation takes into account declared dependencies, the following scenario will come up:
	a) A package.yml specifies a legitimate dependency
	b) Subsequently, a new layer validation is introduced involving the valid dependency
	c) If layer validation flags the dependency as invalid, the dependency will need to be removed and both `layer` and `dependency` violations will be added to the associated package_todo.yml
3) Layer validation against unnecessary dependencies is of limited value. If we're concerned about unnecessary dependencies, we could consider making them dependency validation errors.

Reply:
I'm fine with keeping what you have, but here's what I think:

1. I see things slightly differently here. I don't consider the `dependencies` key to be a property/definition of the dependency checker. Instead, I see `dependencies` as being part of the fundamental properties of a pack, and both the dependency checker and the architectural checker can apply their independent constraints on top of that shared concept.
2. This almost seems like a good thing to me and is conceptually correct – the new architecture constraint would dictate that the dependency is no longer permitted. If the user wants to ignore the subsequent dependency violations associated, they can add the dependency to `ignored_dependencies`.
3. Agreed, it doesn't add much value. I think it's nice in so far as someone can trust that when they look at a package dependencies, it should never violate the architecture.

Ultimately though no big deal... we can try this out and iterate!